ladybug_geometry/intersection2d.py

```diff
@@ -77,6 +79,70 @@ def does_intersection_exist_line2d(A, B):
     return True
 
 
+def intersect_line2d_arc2d(L, C):
+    """Get the intersection between any Ray2D/LineSegment2D and an Arc2D.
+
+    Args:
+        L: A LineSegment2D or Ray2D object.
+        C: An Arc2D object along which the closest point will be determined.
+
+    Returns:
+        A list of 2 Point2D objects if a full intersection exists.
+        A list with a single Point2D object if the line is tangent or intersects
+        only once. None if no intersection exists.
+    """
+    a = L.v.magnitude_squared
+    b = 2 * (L.v.x * (L.p.x - C.c.x) + L.v.y * (L.p.y - C.c.y))
+    c = C.c.magnitude_squared + L.p.magnitude_squared - 2 * C.c.dot(L.p) - C.r ** 2
+    det = b ** 2 - 4 * a * c
+    if det < 0:
+        return None
+    sq = math.sqrt(det)
+    u1 = (-b + sq) / (2 * a)
+    u2 = (-b - sq) / (2 * a)
+    pt1 = Point2D(L.p.x + u1 * L.v.x, L.p.y + u1 * L.v.y) if L._u_in(u1) else None
+    pt2 = Point2D(L.p.x + u2 * L.v.x, L.p.y + u2 * L.v.y) if L._u_in(u2) else None
+
+    if u1 == u2:  # Tangent
+        pt = Point2D(L.p.x + u1 * L.v.x, L.p.y + u1 * L.v.y)
+        return pt if C._pt_in(pt) else None
+
+    pts = [p for p in (pt1, pt2) if p is not None and C._pt_in(p)]
+    return pts if len(pts) != 0 else None
+
+
+def intersect_line2d_infinite_arc2d(L, C):
+    """Get the intersection between an Arc2D and a Ray2D/LineSegment2D extended infinitely.
+
+    Args:
+        L: A LineSegment2D or Ray2D that will be extended infinitely for intersection.
+        C: An Arc2D object along which the closest point will be determined.
+
+    Returns:
+        A list of 2 Point2D objects if a full intersection exists.
+        A list with a single Point2D object if the line is tangent or intersects
+        only once. None if no intersection exists.
+    """
+    a = L.v.magnitude_squared
+    b = 2 * (L.v.x * (L.p.x - C.c.x) + L.v.y * (L.p.y - C.c.y))
+    c = C.c.magnitude_squared + L.p.magnitude_squared - 2 * C.c.dot(L.p) - C.r ** 2
+    det = b ** 2 - 4 * a * c
+    if det < 0:
+        return None
+    sq = math.sqrt(det)
+    u1 = (-b + sq) / (2 * a)
+    u2 = (-b - sq) / (2 * a)
+
+    if u1 == u2:  # Tangent
+        pt = Point2D(L.p.x + u1 * L.v.x, L.p.y + u1 * L.v.y)
+        return pt if C._pt_in(pt) else None
+
+    pt1 = Point2D(L.p.x + u1 * L.v.x, L.p.y + u1 * L.v.y)
+    pt2 = Point2D(L.p.x + u2 * L.v.x, L.p.y + u2 * L.v.y)
+    pts = [p for p in (pt1, pt2) if C._pt_in(p)]
+    return pts if len(pts) != 0 else None
+
+
 def closest_point2d_on_line2d(P, L):
```

Comment:
any reason that we are using uppercase letters for arguments in some of the methods?  -- PEP8 👮

Reply:
Only because that was the case in euclid, which is where all of these intersection functions come from.  I only do this in the `intersect2d` and `intersect3d` modules so it's not too much that needs to be fixed. Still, it's still a relatively big effort compared to this PR so can we merge in this PR and then I will send another fixing this PEP8 issue across the `intersect2d` and `intersect3d` modules?

Reply:
Ah, I see that you approved it.  Ok. I'll merge this one in and then send another PR for the PEP8 👮